p2p/src/net/mock/transport/adapted.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use tokio::{sync::mpsc::UnboundedReceiver, time::timeout};
+
+use crate::{error::P2pError, net::mock::peer::Role, Result};
+
+use super::{MockListener, MockTransport, StreamAdapter};
+
+// How much time is allowed to spend setting up (optionally) encrypted stream.
+const HANDSHAKE_TIMEOUT: Duration = Duration::from_secs(10);
+
+#[derive(Debug)]
+pub struct AdaptedMockTransport<S, T> {
+    stream_adapter: Arc<S>,
```

Comment:
It's probably a bad idea to have this as an Arc. Why not have the stream object owned by the transport layer?

Reply:
`Arc` was removed after recent changes (replacing task spawning).